packages/samples/headless-ssr-commerce/src/lib/navigatorContextProvider.ts

```diff
@@ -0,0 +1,32 @@
+import {NavigatorContext} from '@coveo/headless/ssr-commerce';
+import type {ReadonlyHeaders} from 'next/dist/server/web/spec-extension/adapters/headers';
+
+export class NextJsNavigatorContext implements NavigatorContext {
+  constructor(private headers: ReadonlyHeaders) {}
+
+  get referrer() {
+    return this.headers.get('referer') || this.headers.get('referrer'); // Some browsers use 'referer'
+  }
+
+  get userAgent() {
+    return this.headers.get('user-agent');
+  }
+
+  get location() {
+    return 'TODO:';
```

Comment:
Should we recommend passing / retrieving this information in a custom header (e.g., `x-coveo-location`)?